notebooks/colab_finetuning.ipynb

```diff
@@ -0,0 +1,2024 @@
+        "--lr_scheduler\n",
+        "dual_phase_warmup_constant_cosine_decay\n",
+        "--warmup_steps\n",
+        ".01\n",
+        "--phase2_lr_ratio\n",
+        "0.1\n",
+        "\n",
+        "```"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "16327f53-7beb-412e-a925-12884e66d70b",
+      "metadata": {
+        "id": "16327f53-7beb-412e-a925-12884e66d70b"
+      },
+      "source": [
+        "## Logging arguments"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "132db93c-85e6-4658-a31e-9b103df34cb7",
+      "metadata": {
+        "id": "132db93c-85e6-4658-a31e-9b103df34cb7"
+      },
+      "source": [
+        "We have built in functionality for tracking model training with Weights & Biases. If you have a Weights and Biases account, set the environment variable `WANDB_API_KEY` to your API key and set the flag `--use_wandb` instead of `--no_use_wandb` below.\n",
+        "\n",
+        "```\n",
+        "--no_use_wandb\n",
+        "--wandb_project\n",
+        "metl-target\n",
+        "--wandb_online\n",
+        "--experiment\n",
+        "default\n",
+        "```\n",
+        "\n",
+        "The below argument determines where to place the log directory locally.\n",
+        "```\n",
+        "--log_dir_base\n",
+        "output/training_logs\n",
+        "```"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "53a2fda3-6dfa-46d5-ad3d-3055eda0b29a",
+      "metadata": {
+        "id": "53a2fda3-6dfa-46d5-ad3d-3055eda0b29a"
+      },
+      "source": [
+        "# Running training"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "8d3d8d23-9d54-4888-842d-4fc8fd843b40",
+      "metadata": {
+        "id": "8d3d8d23-9d54-4888-842d-4fc8fd843b40"
+      },
+      "source": [
+        "All the arguments described above are contained in [finetune_avgfp_local.txt](../args/pretrain_avgfp_local.txt), which can be fed directly into [train_target_model.py](train_target_model.py).\n",
+        "\n",
+        "PyTorch Lightning has a built-in progress bar that is convenient for seeing training progress, but it does not display correctly in Jupyter when calling the script with `!python`. We are going to disable the progress bar for by setting the flag `--enable_progress_bar false`. Instead, we implemented a simple print statement to track training progress, which we will enable with the flag `--enable_simple_progress_messages`.\n",
+        "\n",
+        "The [train_target_model.py](../code/train_target_model.py) script can support running on Apple Silicon with acceleration via MPS, but the version of PyTorch used in this environment is slightly outdated and does not support all MPS operations, so MPS support has been disabled. The script will run on GPU via CUDA if available, otherwise it will use CPUs.\n",
+        "\n",
+        "To speed up training for demo purposes, we also override `--max_epochs 50` and `--unfreeze_backbone_at_epoch 25`."
```

Comment:
If you want to run full-length fine-tuning you have to change it, but it's not really described. We can delete it or add a GUI interface for all of these variables so it's a bit more clear.

In the colab one we could probably delete that line about dealing with apple silicon though, since it's just going to run with CUDA